middleware/http/opa/middleware.go

```diff
@@ -98,110 +105,96 @@ func (s *Status) Valid() bool {
 }
 
 // GetHandler returns the HTTP handler provided by the middleware.
-func (m *Middleware) GetHandler(metadata middleware.Metadata) (func(h fasthttp.RequestHandler) fasthttp.RequestHandler, error) {
+func (m *Middleware) GetHandler(metadata middleware.Metadata) (func(next http.Handler) http.Handler, error) {
 	meta, err := m.getNativeMetadata(metadata)
 	if err != nil {
 		return nil, err
 	}
 
-	ctx := context.Background()
-
 	query, err := rego.New(
 		rego.Query("result = data.http.allow"),
 		rego.Module("inline.rego", meta.Rego),
-	).PrepareForEval(ctx)
+	).PrepareForEval(context.Background())
```

Comment:
This is done during the init stages of the component and there's no context. I guess what I can do is to add a timeout here?

Reply:
If I am not wrong each component has an init timeout correct ? If defined can we use some part of that timeout here ?
Since defining a constant timeout might affect init if init timeout is also specified?

Reply:
If we want to rely on the component's init timeout, then I think i need to use a background context. Right now, no context is cascaded. Unless we change the components' interface to cascade context?

TBH I don't even know what the context here is for 😅 No idea what `PrepareForEval` even does.

Reply:
Ideally since all other building blocks support context cascading this should too ? But as a separate pr ? And leave it as background context for now here ?

Reply:
As per conversation offline, given that this is for an init step and callers many levels up don't have a context either, I've changed it to a `context.TODO()` with a 1-min timeout. Should be good enough here.